### PR TITLE
fix: hide Ubuntu Pro bearer token in mirror source URL display

### PR DESCRIPTION
**Pull request by yurii-vasyliev**
## Summary

Follow-up hardening to #663 (LNDENG-4627). That PR composes an Ubuntu Pro mirror's `archiveRoot` as `https://bearer:<token>@esm.ubuntu.com/<service>/ubuntu/`, embedding the bearer token. The mirror **details** view and the **edit** form render `mirror.archiveRoot` verbatim (as link text, an `href`, and a read-only field), so the token would be shown to anyone who can view the mirror's config.

This wires up the existing (previously dead) `getStrippedUrl` helper to strip embedded credentials before display.

## Changes

- Moved `getStrippedUrl` from `AddMirrorForm/helpers.ts` (where it was unused) to `MirrorDetails/helpers.ts`, alongside `getSourceType` — both parse `archiveRoot` and both display sites import from there.
- `MirrorDetails.tsx`: strip credentials from the Source URL link text **and** `href`.
- `EditMirrorForm.tsx`: strip credentials from the read-only Source URL field.
- Added `MirrorDetails/helpers.test.ts` covering the strip and the credential-free no-op.

## Notes

- No-op for non-Pro mirrors (no credentials to strip), so it's safe across all source types.
- The token only appears in `archiveRoot` once #663 merges, so this is effectively a no-op until then — it's defensive display hardening that should land alongside it.
- Whether the API echoes the token back in `archiveRoot` on read wasn't verified from the UI side; this strips client-side regardless.

## Checklist

- [x] Changeset added
- [x] Typecheck + affected tests pass

### CODE REVIEW

**Review by rubinaga — DISMISSED** [review]
LGTM, could consider changing the mock data to include a bearer token in archiveRoot for easier visual testing in the future

File: src/features/mirrors/components/MirrorDetails/helpers.ts [hunk 1]
```diff
@@ -16,3 +16,10 @@ export function getSourceType(archiveRoot: string): string {
       return "Third party";
   }
 }
+
+// Strip any embedded credentials (e.g. the Ubuntu Pro bearer token in
+// `https://bearer:<token>@…`) so they are never shown when displaying an
+// archive root. A no-op for URLs without credentials.
+export function getStrippedUrl(url: string): string {
+  return url.replace(/\/[^\\/@]*@/, "/");
+}
```
[the remark this answers is not in the document]
Comment (gesquivelgaghi) [reply src/features/mirrors/components/MirrorDetails/helpers.ts:112]:
```suggestion
// Strip any embedded credentials (e.g. the Ubuntu Pro bearer token in
// `https://bearer:<token>@…`) so they are never shown when displaying an
// archive root. A no-op for URLs without credentials.
export function getStrippedUrl(url: string): string {
  return url.replace(/\/\/[^\\/@]*@/, "//");
}
```
could also do this to make sure we only get the credential part of the url

[the remark this answers is not in the document]
Comment (yurii-vasyliev) [reply src/features/mirrors/components/MirrorDetails/helpers.ts:112]: Good catch — addressed in 5d2f3f4. I replaced the regex with the `URL` API: `getStrippedUrl` now parses the URL and clears `username`/`password`, falling back to the raw string if it can't be parsed as an absolute URL. This can't misfire on an `@` that's part of the path, and it's consistent with the existing `new URL(archiveRoot)` usage in this file.

@gesquivelgaghi — I went with the URL API rather than the anchored `//`-regex suggestion. Same result for the credential case, but it's more robust against path content and parser edge cases. Thanks for the suggestion!

File: src/features/mirrors/components/MirrorDetails/helpers.test.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        "https://bearer:s3cr3t-token@esm.ubuntu.com/infra/ubuntu/",
+      ),
+    ).toBe("https://esm.ubuntu.com/infra/ubuntu/");
+  });
+
+  it("leaves a credential-free URL unchanged", () => {
+    expect(getStrippedUrl("https://archive.ubuntu.com/ubuntu/")).toBe(
+      "https://archive.ubuntu.com/ubuntu/",
+    );
+  });
```
[the remark this answers is not in the document]
Comment (yurii-vasyliev) [reply src/features/mirrors/components/MirrorDetails/helpers.test.ts]: Added in 5d2f3f4 — there's now a regression test asserting `https://example.com/foo@bar/` is left unchanged (an `@` in the path), plus one for the non-URL fallback.

File: src/features/mirrors/components/MirrorDetails/helpers.ts [hunk 3]
```diff
@@ -94,3 +94,18 @@ export function shouldShowAuthentication(mirror: Mirror): boolean {
 
   return getSourceType(mirror) === MIRROR_TYPE_LABELS.THIRD_PARTY;
 }
+
+// Strip any embedded credentials (e.g. the Ubuntu Pro bearer token in
+// `https://bearer:<token>@…`) so they are never shown when displaying an
+// archive root. A no-op for URLs without credentials; returns the input
+// unchanged if it can't be parsed as an absolute URL.
```
[the remark this answers is not in the document]
Comment (yurii-vasyliev) [reply src/features/mirrors/components/MirrorDetails/helpers.ts]: Fair point — updated the comment in b58a6d7 to drop the "no-op" wording. It now notes that the userinfo is cleared and the URL is otherwise returned in its normalized form (and unchanged if it can't be parsed). The normalization is fine for archive-root display.

File: src/features/mirrors/components/MirrorDetails/MirrorDetails.tsx [hunk 4]
```diff
@@ -151,11 +155,11 @@ const MirrorDetails: FC = () => {
                   label="Source URL"
                   value={
                     <a
-                      href={mirror.archiveRoot}
+                      href={getStrippedUrl(mirror.archiveRoot)}
                       target="_blank"
                       rel="noopener noreferrer"
                     >
-                      {mirror.archiveRoot}
+                      {getStrippedUrl(mirror.archiveRoot)}
```
[the remark this answers is not in the document]
Comment (yurii-vasyliev) [reply src/features/mirrors/components/MirrorDetails/MirrorDetails.tsx:162]: Added a component-level test in b58a6d7 (`MirrorDetails` → "strips embedded credentials from the Source URL"). It renders the new Ubuntu Pro mock mirror (whose `archiveRoot` carries a bearer token), asserts the Source URL link text and `href` are the stripped URL, and asserts the token is not present in the DOM.

File: src/features/mirrors/components/EditMirrorForm/EditMirrorForm.tsx [hunk 5]
```diff
@@ -102,7 +103,7 @@ const EditMirrorForm: FC = () => {
               />
               <ReadOnlyField
                 label="Source URL"
-                value={mirror.archiveRoot}
+                value={getStrippedUrl(mirror.archiveRoot)}
                 tooltipMessage="You can’t change the source URL after the mirror is created."
```
[the remark this answers is not in the document]
Comment (yurii-vasyliev) [reply src/features/mirrors/components/EditMirrorForm/EditMirrorForm.tsx:107]: Added in b58a6d7 (`EditMirrorForm` → "strips embedded credentials from the read-only Source URL"): renders the Ubuntu Pro mock mirror in the edit form and asserts the read-only Source URL shows the stripped value with the token absent from the DOM.

**Comment by gesquivelgaghi** [comment]
the thing about this is that the backend is already doing it. They mentioned it in the squad meeting yesterday and I just found this ticket [LNDENG-4632](https://warthogs.atlassian.net/browse/LNDENG-4632) now in "To be deployed" status, so it seems like the work is already done 

[LNDENG-4632]: https://warthogs.atlassian.net/browse/LNDENG-4632?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by yurii-vasyliev** [comment]
Closing this — it's no longer needed.

The Ubuntu Pro bearer token is now stripped **server-side**, so the mirror source URL already arrives stripped. The client-side `getStrippedUrl` approach this PR takes (wiring it into `EditMirrorForm` / `MirrorDetails`) is therefore redundant.

The leftover unused `getStrippedUrl` helper on `main` is being removed separately in #699.
